java/vector/src/main/java/org/apache/arrow/vector/util/Validator.java

```diff
@@ -33,101 +38,140 @@
-        }
-        List<FieldVector> arrowVectors = root1.getFieldVectors();
-        List<FieldVector> jsonVectors = root2.getFieldVectors();
-        if (arrowVectors.size() != jsonVectors.size()) {
-            throw new IllegalArgumentException("Different column count:\n" + arrowVectors.size() + "\n" + jsonVectors.size());
-        }
-        for (int i = 0; i < arrowVectors.size(); i++) {
-            Field field = root1.getSchema().getFields().get(i);
-            FieldVector arrowVector = arrowVectors.get(i);
-            FieldVector jsonVector = jsonVectors.get(i);
-            int valueCount = arrowVector.getAccessor().getValueCount();
-            if (valueCount != jsonVector.getAccessor().getValueCount()) {
-                throw new IllegalArgumentException("Different value count for field " + field + " : " + valueCount + " != " + jsonVector.getAccessor().getValueCount());
-            }
-            for (int j = 0; j < valueCount; j++) {
-                Object arrow = arrowVector.getAccessor().getObject(j);
-                Object json = jsonVector.getAccessor().getObject(j);
-                if (!equals(field.getType(), arrow, json)) {
-                    throw new IllegalArgumentException(
-                            "Different values in column:\n" + field + " at index " + j + ": " + arrow + " != " + json);
-                }
-            }
-        }
+  /**
+   * Validate two Dictionary encodings and dictionaries with id's from the encodings
+   */
+  public static void compareDictionaries(List<DictionaryEncoding> encodings1, List<DictionaryEncoding> encodings2, DictionaryProvider provider1, DictionaryProvider provider2) {
+
+    if (encodings1.size() != encodings2.size()) {
+      throw new IllegalArgumentException("Different dictionary encoding count:\n" + encodings1.size() + "\n" + encodings2.size());
     }
 
-    static boolean equals(ArrowType type, final Object o1, final Object o2) {
-        if (type instanceof ArrowType.FloatingPoint) {
-            ArrowType.FloatingPoint fpType = (ArrowType.FloatingPoint) type;
-            switch (fpType.getPrecision()) {
-                case DOUBLE:
-                    return equalEnough((Double)o1, (Double)o2);
-                case SINGLE:
-                    return equalEnough((Float)o1, (Float)o2);
-                case HALF:
-                default:
-                    throw new UnsupportedOperationException("unsupported precision: " + fpType);
-            }
-        } else if (type instanceof ArrowType.Binary) {
-            return Arrays.equals((byte[]) o1, (byte[]) o2);
-        }
+    Comparator<DictionaryEncoding> comp = new Comparator<DictionaryEncoding>() {
+      @Override
+      public int compare(DictionaryEncoding enc1, DictionaryEncoding enc2) {
+        return enc1.getId() < enc2.getId() ? -1 : 1;
+      }
+    };
+
+    // Order of encodings not important
+    Collections.sort(encodings1, comp);
+    Collections.sort(encodings2, comp);
```

Comment:
we should probably avoid sorting the input collection - the user might not be expecting that. instead, create a new collection and sort that.